biochemist-python/chapters/File_Parsing.ipynb

```diff
@@ -0,0 +1,602 @@
+    "print(protein_file)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You should see something similar to the following:\n",
+    "\n",
+    "data/PDB_files/4eyr.pdb"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here, we have specified that our filepath contains the 'data' and 'PDB_files' directory, and the `os.path` module has made this into a filepath that is usable by our system. If you are on Windows, you will instead see that a forward slash is used."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Absolute and relative paths\n",
+    "File paths can be *absolute*, or *relative*.\n",
+    "\n",
+    "A relative file path gives the location relative to the directory we are in. Thus, if we are in the `biochemist-python` directory, the relative filepath for the `4eyr.pdb` file would be `chapters/data/PDB_files/4eyr.pdb`\n",
+    "\n",
+    "An absolute filepath gives the complete path to a file. This could file path could be used from anywhere on a computer, and would access the same file. For example, the absolute filepath to the `4eyr.pdb` file on a Mac might be `/Users/YOUR_USER_NAME/Desktop/python-scripting-biochemistry/biochemist-python/chapters/data/PDB_files/4eyr.pdb`. You can get the absolute path of a file using `os.path.abspath(path)`, where path is the relative path of the file."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```{note}\n",
+    "We are working with the `os.path` module here, and this is how you will see people handle file paths in most Python code. However, as of Python 3.6, there is also a `pathlib` module in the Python standard library that can be used to represent and manipulate filepaths. `os.path` works with filepaths as strings, while in the `pathlib` module, paths are objects. A good overview of the pathlib module can be found [here](https://realpython.com/python-pathlib/).\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Reading a file\n",
+    "\n",
+    "In Python, there are many ways to read in information from a text file.  The best method to use depends on the type of data and the type of analysis you are performing.  If you have a file with lots of different types of information, text and numbers, with different types of formatting, the most generic way to read in information is the `readlines()` function.  Before you can read in a file, you have to open the file using the file path we defined above. This will create a file object, or filehandle.  The file we will be analyzing in this example is a [PDB web site](https://www.rcsb.org/structure/4EYR) file for an HIV protease complex with the inhibitor Ritonavir."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "outfile = open(protein_file,\"r\")\n",
```

Comment:
The preferred method to do this would actually be to do 

    with open(protein_file, "r") as f:
        data = f.readlines()
     
We have been considering switching this in the python scripting lesson for a while. If this syntax is used, you do not have to remember to close the file and Python does it automatically.

Reply:
Done